Source/FullScreenPokemon.ts

```diff
@@ -2511,6 +2524,8 @@ module FullScreenPokemon {
                     this.StateHolder.addChange(other.id, "sight", undefined);
                 }
             }
+
+            this.autoSave();
```

Comment:
This get called if we're starting a battle with the trainer, from the `if (other.trainer && !(<IEnemy>other).alreadyBattled)` block. I think this should be if that isn't true, yes?